lib/streamlit/elements/skeleton.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, cast
+
+from streamlit.proto.Skeleton_pb2 import Skeleton as SkeletonProto
+from streamlit.runtime.metrics_util import gather_metrics
+
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+
+# Works just like an `empty` except animated by default.
+class SkeletonMixin:
+    @gather_metrics("skeleton")
+    def skeleton(self, height: int = 107) -> "DeltaGenerator":
+        """Insert a single-element container which displays a "skeleton" animation.
+
+        Inserts a container into your app that can be used to hold a single element.
+        This allows you to, for example, remove elements at any point, or replace
+        several elements at once (using a child multi-element container).
+
+        To insert/replace/clear an element on the returned container, you can
+        use "with" notation or just call methods directly on the returned object.
+        See some of the examples below.
+
+        By default skeletons have a display height of 230pt, comprised of a 123pt header
+        and a 107pt adjustable-height block.
+
+        Parameters
+        ----------
+        height: int
+            An integer representing the height in CSS points of the adjustable-height block.
+
+
+        Examples
+        --------
+        st.skeleton is an st.empty with a future promise of additional content (typically one piece).
+        The primary use case is to enable users to build Streamlit applications which
+        render UI first and content second.
+
+        This means you can replace
+
+        >>> # Draw UI
+        >>> st.write("Census Data")
+        >>> census_data_placeholder = st.empty()  # The empty is a collapsed element
+        >>> year = st.selectbox("Select Year", options=[2022, 2023, 2024])
+        >>>
+        >>> # Get data from source
+        >>> # assume function call takes a long time
```

Comment:
nit: this comment is very slightly different from the corresponding one in the skeleton example below